dist/profile/templates/buildmaster/casc/clouds.yaml.erb

```diff
@@ -0,0 +1,171 @@
+---
+jenkins:
+  clouds:
+  <%- if @azure_vm_agents && !@azure_vm_agents.empty? -%>
+  - azureVM:
+      azureCredentialsId: "azure-credentials"
+      cloudName: "Azure"
+      configurationStatus: "pass"
+      deploymentTimeout: 1200
+      <%- $maxAzureVms = 0; @azure_vm_agents.each do |agent| -%>
+      <%- $maxAzureVms = agent["maxInstances"] + $maxAzureVms -%>
+      <%- end -%>
+      maxVirtualMachinesLimit: <%= $maxAzureVms %>
+      existingResourceGroupName: "eastus-cijenkinsio"
+      vmTemplates:
+      <%- @azure_vm_agents.each do |agent| -%>
+      - agentLaunchMethod: "SSH"
+        agentWorkspace: "<%= @agents_setup[agent["os"]]["agentDir"] %>"
+        credentialsId: "jenkinsvmagents-userpass"
+        diskType: "managed"
+        doNotUseMachineIfInitFails: true
+        enableMSI: false
+        enableUAMI: false
+        ephemeralOSDisk: true
+        existingStorageAccountName: "<%= agent["storageAccount"] %>"
+        storageAccountType: "Standard_LRS"
+        imageReference:
+          galleryImageDefinition: "<%= agent["imageDefinition"] %>"
+          galleryImageVersion: "<%= @agent_images["azure_vms_gallery_image"]["version"] %>"
+          galleryName: "prod_packer_images"
+          galleryResourceGroup: "prod-packer-images"
+          gallerySubscriptionId: "<%= @agent_images["azure_vms_gallery_image"]["subscription_id"] %>"
+        imageTopLevelType: "advanced"
+        installDocker: false
+        installGit: false
+        installMaven: false
+        javaPath: "java"
+        labels: "<%= agent["os"] %> <%= agent["architecture"] %> azure vm <%= agent["labels"].join(' ') %>"
+        location: "<%= agent["location"] %>"
+        maximumDeploymentSize: <%= agent["maxInstances"] %>
+        noOfParallelJobs: 1
+        osDiskSize: 0
+        osType: "<%= agent["os"] == "windows" ? 'Windows' : 'Linux' %>"
+        preInstallSsh: false
+        retentionStrategy:
+          azureVMCloudRetentionStrategy:
+            idleTerminationMinutes: 5
+        shutdownOnIdle: false
+        templateDesc: "Dynamically provisioned <%= agent["description"] %> machine"
+        templateDisabled: false
+        templateName: "<%= agent["name"] %>"
+        usageMode: "Use this node as much as possible"
+        usePrivateIP: false
+        virtualMachineSize: "<%= agent["instanceType"] %>"
+      <%- end -%>
+  <%- end -%>
+  - aci:
```

Comment:
shouldn't ACI be guarded by an if statement too?

Reply:
Yes, but I like to split my work as much as possible: I wanted to do it on a subsequent PR

Reply:
Oh nevermind: it has to be moved on azure.ci, good catch! thanks

Reply:
done